### PR TITLE
chore: Bump Neo4j version from 3.3 to 3.5

### PR DESCRIPTION
**Pull request by madison-ookla**
### Summary of Changes

This PR bumps the version of Neo4j specified in the relevant docker-compose files from 3.3 to 3.5. This change was initiated because an attempt to use modern Neo4j tooling (e.g. Neo4j Desktop Canary) to connect to a non-local Neo4j database running version 3.3 failed due to driver incompatibilities. Additionally, version 3.3 is [officially unsupported as of April 28, 2019](https://neo4j.com/developer/kb/neo4j-supported-versions/) (3.5 will be supported until November 28, 2021). Version 3.5 is a compatibility release which allows the newer v4 drivers to be used when connecting, so this resolves the aforementioned issue on modern tooling. Becuase it is a compat release, no Amundsen packages/code need to be changed for this upgrade. There were a few directory changes that needed to be made from 3.3 -> 3.5; these have been added to this PR.

This is related to the work in amundsen-io/amundsendatabuilder#447.

I have tested this locally and ensured that data persists in the specified volumes after restarts.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [x] PR title addresses the issue accurately and concisely, including a title prefix.
- [x] PR includes a summary of changes.
- [x] My commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

**Review by feng-tao — APPROVED** [review]

**Comment by dorianj** [comment]
When I try this out locally with a vanilla `docker-compose -f docker-amundsen.yml up`, I get this:

```
neo4j_amundsen      | 2021-03-10 19:10:46.250+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.". Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      | org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:187)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:124)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
neo4j_amundsen      |   at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:180)
neo4j_amundsen      |   ... 3 more
neo4j_amundsen      | Caused by: java.lang.RuntimeException: Error starting org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory, /neo4j/data/databases
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:232)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:148)
neo4j_amundsen      |   at org.neo4j.server.database.CommunityGraphFactory.newGraphDatabase(CommunityGraphFactory.java:41)
neo4j_amundsen      |   at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:90)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 5 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@5be46f9d' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:227)
neo4j_amundsen      |   ... 9 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.StoreLockException: Unable to create path for store dir: /neo4j/data/databases. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:79)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:61)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 11 more
neo4j_amundsen      | Caused by: java.io.IOException: Unable to create directory path [/neo4j/data/databases] for Neo4j store.
neo4j_amundsen      |   at org.neo4j.io.fs.DefaultFileSystemAbstraction.mkdirs(DefaultFileSystemAbstraction.java:120)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:73)
neo4j_amundsen      |   ... 14 more
neo4j_amundsen      | 2021-03-10 19:10:46.262+0000 INFO  Neo4j Server shutdown initiated by request
neo4j_amundsen exited with code 1
```

Any advice or something I'm doing wrong?

**Comment by madison-ookla** [comment]
> When I try this out locally with a vanilla `docker-compose -f docker-amundsen.yml up`, I get this:
> 
> ```
> neo4j_amundsen      | 2021-03-10 19:10:46.250+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.". Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
> neo4j_amundsen      | org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
> neo4j_amundsen      |   at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
> neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:187)
> neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:124)
> neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
> neo4j_amundsen      |   at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
> neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
> neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:180)
> neo4j_amundsen      |   ... 3 more
> neo4j_amundsen      | Caused by: java.lang.RuntimeException: Error starting org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory, /neo4j/data/databases
> neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:232)
> neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:148)
> neo4j_amundsen      |   at org.neo4j.server.database.CommunityGraphFactory.newGraphDatabase(CommunityGraphFactory.java:41)
> neo4j_amundsen      |   at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:90)
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
> neo4j_amundsen      |   ... 5 more
> neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@5be46f9d' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
> neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:227)
> neo4j_amundsen      |   ... 9 more
> neo4j_amundsen      | Caused by: org.neo4j.kernel.StoreLockException: Unable to create path for store dir: /neo4j/data/databases. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
> neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
> neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:79)
> neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:61)
> neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
> neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
> neo4j_amundsen      |   ... 11 more
> neo4j_amundsen      | Caused by: java.io.IOException: Unable to create directory path [/neo4j/data/databases] for Neo4j store.
> neo4j_amundsen      |   at org.neo4j.io.fs.DefaultFileSystemAbstraction.mkdirs(DefaultFileSystemAbstraction.java:120)
> neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:73)
> neo4j_amundsen      |   ... 14 more
> neo4j_amundsen      | 2021-03-10 19:10:46.262+0000 INFO  Neo4j Server shutdown initiated by request
> neo4j_amundsen exited with code 1
> ```
> 
> Any advice or something I'm doing wrong?

There should be a few configuration messages before this one, do you mind including them? Here are the ones I receive when I run the same command:
```
neo4j_amundsen      | Active database: graph.db
neo4j_amundsen      | Directories in use:
neo4j_amundsen      |   home:         /var/lib/neo4j
neo4j_amundsen      |   config:       /var/lib/neo4j/conf
neo4j_amundsen      |   logs:         /logs
neo4j_amundsen      |   plugins:      /var/lib/neo4j/plugins
neo4j_amundsen      |   import:       NOT SET
neo4j_amundsen      |   data:         /var/lib/neo4j/data
neo4j_amundsen      |   certificates: /var/lib/neo4j/certificates
neo4j_amundsen      |   run:          /var/lib/neo4j/run
```

The fact that you're seeing `Unable to create path for store dir: /neo4j/data/databases.` leads me to believe that the `/data` directory config (set [here](https://github.com/amundsen-io/amundsen/pull/953/files#diff-e77ace1447b4b27ea26f7107da26069abdc5c091fbec233c10f6daa6d6ab3238R19) somehow isn't being respected or isn't present.

**Comment by dorianj** [comment]
Sure! (sorry, I should have included it before, but was mixed with other stuff)
```
dorian@zarya amundsen % docker-compose -f docker-amundsen.yml up neo4j
Recreating neo4j_amundsen ... done
Attaching to neo4j_amundsen
neo4j_amundsen      | Warning: Some files inside "/data" are not writable from inside container. Changing folder owner to neo4j.
neo4j_amundsen      | Changed password for user 'neo4j'.
neo4j_amundsen      | Active database: amundsen.db
neo4j_amundsen      | Directories in use:
neo4j_amundsen      |   home:         /var/lib/neo4j
neo4j_amundsen      |   config:       /var/lib/neo4j/conf
neo4j_amundsen      |   logs:         /logs
neo4j_amundsen      |   plugins:      /var/lib/neo4j/plugins
neo4j_amundsen      |   import:       /var/lib/neo4j/import
neo4j_amundsen      |   data:         /neo4j/data
neo4j_amundsen      |   certificates: /var/lib/neo4j/certificates
neo4j_amundsen      |   run:          /var/lib/neo4j/run
neo4j_amundsen      | Starting Neo4j.
neo4j_amundsen      | 2021-03-10 20:09:12.254+0000 WARN  Unknown config option: dbms.shell.enabled
neo4j_amundsen      | 2021-03-10 20:09:12.271+0000 WARN  dbms.active_database is deprecated.
neo4j_amundsen      | 2021-03-10 20:09:12.438+0000 INFO  ======== Neo4j 3.5.26 ========
neo4j_amundsen      | 2021-03-10 20:09:12.521+0000 INFO  Starting...
neo4j_amundsen      | 2021-03-10 20:09:20.766+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.". Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      | org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:187)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:124)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
neo4j_amundsen      |   at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:180)
neo4j_amundsen      |   ... 3 more
neo4j_amundsen      | Caused by: java.lang.RuntimeException: Error starting org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory, /neo4j/data/databases
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:232)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:148)
neo4j_amundsen      |   at org.neo4j.server.database.CommunityGraphFactory.newGraphDatabase(CommunityGraphFactory.java:41)
neo4j_amundsen      |   at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:90)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 5 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@5be46f9d' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:227)
neo4j_amundsen      |   ... 9 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.StoreLockException: Unable to create path for store dir: /neo4j/data/databases. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:79)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:61)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 11 more
neo4j_amundsen      | Caused by: java.io.IOException: Unable to create directory path [/neo4j/data/databases] for Neo4j store.
neo4j_amundsen      |   at org.neo4j.io.fs.DefaultFileSystemAbstraction.mkdirs(DefaultFileSystemAbstraction.java:120)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:73)
neo4j_amundsen      |   ... 14 more
neo4j_amundsen      | 2021-03-10 20:09:20.776+0000 INFO  Neo4j Server shutdown initiated by request
neo4j_amundsen exited with code 1
```


It's possible this could be caused because I had an old neo4j container, if that's the case, what do I need to do to reset my environment? (and if that's the case, we'll need a migration path)

**Comment by madison-ookla** [comment]
Hmm, this line is still confusing to me:
```
neo4j_amundsen      |   data:         /neo4j/data
```

If you haven't already, do you mind taking down and back up the entire stack? (e.g. `docker-compose -f docker-amundsen.yml down` then `docker-compose -f docker-amundsen.yml up`)

**Comment by dorianj** [comment]
Sure:
```
dorian@zarya amundsen % docker-compose -f docker-amundsen.yml down
Removing amundsenfrontend ... done
Removing amundsenmetadata ... done
Removing amundsensearch   ... done
Removing neo4j_amundsen   ... done
Removing es_amundsen      ... done
Removing network amundsen_amundsennet
dorian@zarya amundsen % docker-compose -f docker-amundsen.yml up neo4j         
Creating neo4j_amundsen ... done
Attaching to neo4j_amundsen
neo4j_amundsen      | Changed password for user 'neo4j'.
neo4j_amundsen      | Active database: amundsen.db
neo4j_amundsen      | Directories in use:
neo4j_amundsen      |   home:         /var/lib/neo4j
neo4j_amundsen      |   config:       /var/lib/neo4j/conf
neo4j_amundsen      |   logs:         /logs
neo4j_amundsen      |   plugins:      /var/lib/neo4j/plugins
neo4j_amundsen      |   import:       /var/lib/neo4j/import
neo4j_amundsen      |   data:         /neo4j/data
neo4j_amundsen      |   certificates: /var/lib/neo4j/certificates
neo4j_amundsen      |   run:          /var/lib/neo4j/run
neo4j_amundsen      | Starting Neo4j.
neo4j_amundsen      | 2021-03-10 23:41:30.879+0000 WARN  Unknown config option: dbms.shell.enabled
neo4j_amundsen      | 2021-03-10 23:41:30.898+0000 WARN  dbms.active_database is deprecated.
neo4j_amundsen      | 2021-03-10 23:41:31.114+0000 INFO  ======== Neo4j 3.5.26 ========
neo4j_amundsen      | 2021-03-10 23:41:31.206+0000 INFO  Starting...
neo4j_amundsen      | 2021-03-10 23:41:40.243+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.". Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      | org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:187)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:124)
neo4j_amundsen      |   at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:91)
neo4j_amundsen      |   at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.database.LifecycleManagingDatabase@51fadaff' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:180)
neo4j_amundsen      |   ... 3 more
neo4j_amundsen      | Caused by: java.lang.RuntimeException: Error starting org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory, /neo4j/data/databases
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:232)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.newFacade(GraphDatabaseFacadeFactory.java:148)
neo4j_amundsen      |   at org.neo4j.server.database.CommunityGraphFactory.newGraphDatabase(CommunityGraphFactory.java:41)
neo4j_amundsen      |   at org.neo4j.server.database.LifecycleManagingDatabase.start(LifecycleManagingDatabase.java:90)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 5 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter@5be46f9d' was successfully initialized, but failed to start. Please see the attached cause exception "Unable to create directory path [/neo4j/data/databases] for Neo4j store.".
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j_amundsen      |   at org.neo4j.graphdb.facade.GraphDatabaseFacadeFactory.initFacade(GraphDatabaseFacadeFactory.java:227)
neo4j_amundsen      |   ... 9 more
neo4j_amundsen      | Caused by: org.neo4j.kernel.StoreLockException: Unable to create path for store dir: /neo4j/data/databases. Please ensure no other process is using this database, and that the directory is writable (required even for read-only access)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.storeLockException(StoreLocker.java:116)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:79)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.GlobalStoreLocker.checkLock(GlobalStoreLocker.java:61)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLockerLifecycleAdapter.start(StoreLockerLifecycleAdapter.java:36)
neo4j_amundsen      |   at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j_amundsen      |   ... 11 more
neo4j_amundsen      | Caused by: java.io.IOException: Unable to create directory path [/neo4j/data/databases] for Neo4j store.
neo4j_amundsen      |   at org.neo4j.io.fs.DefaultFileSystemAbstraction.mkdirs(DefaultFileSystemAbstraction.java:120)
neo4j_amundsen      |   at org.neo4j.kernel.internal.locker.StoreLocker.checkLock(StoreLocker.java:73)
neo4j_amundsen      |   ... 14 more
neo4j_amundsen      | 2021-03-10 23:41:40.247+0000 INFO  Neo4j Server shutdown initiated by request
neo4j_amundsen exited with code 1
```

**Comment by madison-ookla** [comment]
@dorianj and I did some offline debugging and he identified the cause of this error as the config living in `example/docker/neo4j/conf` within this repo. I had been testing this by moving the docker-compose file _out_ of the repo (I didn't want existing volume names to clash 🤦‍♀️ ) so my instance did not pick up that config at all. I'll be going through this a bit more today to see what, if anything, in the supplied config is still needed.

File: docs/installation-aws-ecs/docker-ecs-amundsen.yml [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
       environment:
         - NEO4J_AUTH=neo4j/test
-        # These dont seem to be working though!
-        - NEO4J_dbms.active_database=amundsen.db
-        - NEO4J_dbms.directories.data=/neo4j/data
-        - NEO4J_dbms.directories.logs=/var/log/neo4j
-        - NEO4J_dbms.directories.import=/var/lib/neo4j/import
-        - NEO4J_dbms.security.auth_enabled=false
-        - NEO4J_dbms.connectors.default_listen_address=0.0.0.0
+        - NEO4J_dbms_active__database=amundsen.db
```
Comment (madison-ookla) [inline docs/installation-aws-ecs/docker-ecs-amundsen.yml:8]: I corrected the notation on these so they're read by Neo4j appropriately

File: example/docker/neo4j/conf/neo4j.conf [hunk 2]
```diff
@@ -228,13 +228,6 @@ dbms.connector.https.enabled=true
 # Retention policy for transaction logs needed to perform recovery and backups.
 dbms.tx_log.rotation.retention_policy=1 days
 
-# Enable a remote shell server which Neo4j Shell clients can log in to.
```
Comment (madison-ookla) [inline example/docker/neo4j/conf/neo4j.conf:231]: These configs are no longer recognized in v3.5

File: docs/installation-aws-ecs/docker-ecs-amundsen.yml (hunk 1 above)
Comment (dorianj) [reply docs/installation-aws-ecs/docker-ecs-amundsen.yml:8]: nice catch

**Review by dorianj — APPROVED** [review]
I tried it out, and it works! It also works fine if you go back in time, check out 3.3, check this out, and re-start the container... so I think we don't need to notify anyone. 

Thanks for paving the way towards 4.0 support ! 😃

**Pull request merged**